processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -349,6 +367,39 @@ public static HumanReadableBytes getAsHumanReadableBytes(
     throw badTypeException(key, "a human readable number", value);
   }
 
+  /**
+   * Insert, update or remove a single key to produce an overridden context.
+   * Leaves the original context unchanged.
+   *
+   * @param context context to override
+   * @param key     key to insert, update or remove
+   * @param value   if {@code null}, remove the key. Otherwise, inert or replace
+   *                the key.
+   * @return a new context map
+   */
+  public static Map<String, Object> override(
+      final Map<String, Object> context,
+      final String key,
+      final Object value
+  )
+  {
+    Map<String, Object> overridden = new TreeMap<>();
```

Comment:
Any special reason to use `TreeMap` instead of `HashMap`?

Reply:
Another good catch. Native queries are serialized. It is handy if that serialization is deterministic. A `HashMap` will return its key/value pairs in random order, but `TreeMap` returns them in sorted order.

Given that, the right place to create the `TreeMap` is in `QueryContext` itself. Here, I lazily copied what was done in another similar method. Since `QueryContext` now copies the context into a `TreeMap`, there is no need to use one here as well.